bin/pkg.ml

```diff
@@ -259,6 +216,43 @@ module Lock = struct
                        (Loc.to_file_colon_line c.loc))))
   end
 
+  (* Add the opam system variables derived from the current system into a
+     solver environment configured in a build context. If variables derived
+     from the current system are also present in the solver environment
+     configured in the build context, an error is raised unless they both have
+     the same value (in which case that value is used). *)
+  let merge_current_system_bindings_into_solver_env_from_context ~context_name
+      ~solver_env_from_context ~sys_bindings_from_current_system =
+    let sys =
+      match
+        Dune_pkg.Solver_env.Sys_var.Bindings.union
+          solver_env_from_context.Dune_pkg.Solver_env.sys
+          sys_bindings_from_current_system
+      with
+      | Ok solver_env -> solver_env
+      | Error
+          (`Var_in_both_with_different_values
+            (var, value_from_context, value_from_system)) ->
+        User_error.raise
+          [ Pp.textf "Can't create solver environment for context: %s"
```

Comment:
Tiny nitpick but text should probably be `Can't create solver environment for context %s`, that is without the colon as it reads weirdly.